packages/time/lib/cso.ts

```diff
@@ -413,6 +413,21 @@ export const findNumCyclesInPastMaturityExists = (
   );
 };
 
+/**
+ * isHalfMonth
+ *
+ * Determine if the current cycle is a half month cycle
+ * @param {string} eventId current time
+ * @returns {boolean} whether the current cycle is a half month
+ */
+export const isHalfMonth = (eventId: string): boolean => {
+  const startDate = getParamsFromCsoEventId(eventId).startDate;
+  return (
+    startDate.getTime() ===
+    getCsoEventDates(startDate).tradingOpenHalfMonth.getTime()
```

Comment:
Might be good to check if this value is `tradingOpenHalfMonth` or `halfMonthEntryClosed`

(we recently switched from half month entry closed to trading open half month, and it would be good to future proof)

Reply:
oh interesting, just updated that